source/features/repo-age.tsx

```diff
@@ -4,50 +4,86 @@ import cache from 'webext-storage-cache';
 import select from 'select-dom';
 import repoIcon from 'octicon/repo.svg';
 import elementReady from 'element-ready';
+import fetchDom from '../libs/fetch-dom';
 import features from '../libs/features';
-import * as api from '../libs/api';
-import {getRepoGQL, getRepoURL} from '../libs/utils';
+import {getRepoURL} from '../libs/utils';
 
 const dateFormatter = new Intl.DateTimeFormat('en-US', {
 	year: 'numeric',
 	month: 'long',
 	day: 'numeric'
 });
 
-const getRepoCreationDate = cache.function(async (): Promise<string> => {
-	const {repository} = await api.v4(`
-		repository(${getRepoGQL()}) {
-			createdAt
-		}
-	`);
+type OldestCommitDetails = {
+	url: string;
+	datetime: string;
+};
 
-	return repository.createdAt;
+const getOldestCommitDetails = cache.function(async (): Promise<OldestCommitDetails | false> => {
+	const lastCommitHash = (await elementReady<HTMLAnchorElement>('.commit-tease-sha', {stopOnDomReady: false}))!.href.split('/').pop()!;
+	const commitsCount = Number(select('li.commits .num')!.textContent!.replace(',', ''));
+
+	if (commitsCount === 0) {
+		return false;
+	}
+
+	if (commitsCount === 1) {
+		return {
+			url: `/${getRepoURL()}/commit/${lastCommitHash}`,
+			datetime: select('.commit-tease-sha + span relative-time')!.getAttribute('datetime')!
+		};
+	}
+
+	const oldestCommit = await fetchDom(
+		`${getRepoURL()}/commits?after=${lastCommitHash}+${commitsCount - 2}`, 'ol:last-child > li.commits-list-item'
+	);
+
+	return {
+		url: select<HTMLAnchorElement>('.commit-title [href*="/commit/"]', oldestCommit)!.href,
+		datetime: select('relative-time', oldestCommit)!.getAttribute('datetime')!
+	};
 }, {
+	isExpired: (cachedValue: OldestCommitDetails | false | string) => !cachedValue || typeof cachedValue === 'string',
```

Comment:
If `isExpired` returns true, the cache is discarded/ignored.

If you include `!cachedValue`, you're saying that a `false` value is expired, i.e. "discard any cached `false` value`.


```suggestion
	isExpired: (cachedValue: OldestCommitDetails | false | string) => typeof cachedValue === 'string',
```

Reply:
`false` will only be cached when there were no commits previously. We again need to check if commits are added. If yes, we try to get the first commit and date otherwise we return `false`. Correct?

Reply:
> We again need to check if commits are added

If you want to check _again_, then it shouldn't be cached in the first place the function should return `undefined`

If you store `false`, every time you open the Repo tab you get 3 `storage` calls:

1. get (returns `false`)
2. delete (due to `isExpiring`)
3. set (sets `false`)

If you don't store it at all, you get:

1. get

Reply:
Thanks for discussing this with me btw, I know it can get frustrating

Reply:
Returning `undefined` would be a better way but seems like types are incompatible. `cache.function` doesn't allow `undefined` to be returned. I think, It should be allowed. The types for both can be made different. I mean returned values from the function and cached values. What do you think?

Reply:
v2.3.0 should do it: https://github.com/fregante/webext-storage-cache/pull/14